### PR TITLE
fix: false positives when adjusting various settings

### PR DESCRIPTION
**Pull request by EsadCetiner**
Fixes various false positives when adjusting user settings, or administrator settings.

### CODE REVIEW

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 1]
```diff
@@ -1201,6 +1201,65 @@ SecRule REQUEST_FILENAME "@rx /apps/recognize/admin/settings/(?:tensorflow\.(?:c
     ctl:ruleRemoveTargetById=932160;ARGS:json.value,\
     setvar:'tx.allowed_methods=%{tx.allowed_methods} PUT'"
 
+# When configuration oauth 2 clients under Administration --> Security --> OAuth 2.0 clients
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:1204]:
```suggestion
# When configuring OAuth 2 clients under Administration --> Security --> OAuth 2.0 clients
```

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 2]
```diff
@@ -1201,6 +1201,65 @@ SecRule REQUEST_FILENAME "@rx /apps/recognize/admin/settings/(?:tensorflow\.(?:c
...
+SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/files_retention/api/v[0-9]/retentions/[0-9]+$" \
+    "id:9508629,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} DELETE'"
+
+# When enabling/disabling Microsoft/Gmail mail integration via oauth 2 under Settings --> Administration
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:1227]:
```suggestion
# When enabling / disabling Microsoft / Gmail email integration via OAuth 2 under Settings --> Administration 
```

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 3]
```diff
@@ -1201,6 +1201,65 @@ SecRule REQUEST_FILENAME "@rx /apps/recognize/admin/settings/(?:tensorflow\.(?:c
...
+# Settings --> Administration --> Log Reader
+SecRule REQUEST_FILENAME "@endsWith /apps/logreader/poll" \
+    "id:9508631,\
+    phase:2,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    chain"
+    SecRule ARGS:lastReqId "@rx ^[a-zA-Z0-9]+$" \
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:1248]:
```suggestion
    SecRule ARGS:lastReqId "@rx (?i)^[a-z0-9]+$" \
```

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 4]
```diff
@@ -1283,7 +1342,12 @@ SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]+\.php/apps/user_status/api/v[0-9]+/use
 #
 
 # Allow users to be deleted and user profile settings to be updated
-SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]+\.php/cloud/users/" \
+# Matches:
+#  /ocs/v2.php/cloud/user-group/username
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:1346]: Make it clear that `user-group` is a place holder:
```suggestion
#  /ocs/v2.php/cloud/<user group>/username
```

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 5]
```diff
@@ -1283,7 +1342,12 @@ SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]+\.php/apps/user_status/api/v[0-9]+/use
 #
 
 # Allow users to be deleted and user profile settings to be updated
-SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]+\.php/cloud/users/" \
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:1286]: Do we need to retain the match of `.../users/` for backwards compatibility?

**Review by theseion — CHANGES_REQUESTED** [review]
Great work!

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 3 above)
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf:1248]: don't they both do the same thing, what's the reasoning for the change?

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 5 above)
Comment (theseion) [reply plugins/nextcloud-rule-exclusions-before.conf:1286]: Thanks, good enough for me.

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 3 above)
Comment (theseion) [reply plugins/nextcloud-rule-exclusions-before.conf:1248]: Yes, they're equivalent. The latter form is just more in line with the other rules in CRS.

**Review by theseion — APPROVED** [review]

**Comment by theseion** [comment]
I'll have to fix the test setup before I can merge...

**Comment by theseion** [comment]
I've fixed the test setup but GH won't pull the new changes in when rerunning the actions, unfortunately. I'll merge, then the merge validation will show any issues.

**Pull request merged**